File: 2023/1/python/part1.py

with open('../data/input.txt', 'r') as f:
    lines = f.read().strip().split('\n')


def get_digits(string):
    for char in string:
        if char.isdigit():
            return char


def main():
    calibration_values = []
    for line in lines:
        first = get_digits(line)
        last = get_digits(line[::-1])

        calibration_values.append(int(first + last))

    print(sum(calibration_values))

if __name__ == '__main__':
    main()

File: 2023/1/golang/part1.go
package main

import (
    "os"
    "log"
    "fmt"
    "strings"
    "unicode"
)

func main() {
    b, err := os.ReadFile("../data/test_1.txt")
    if err != nil { log.Fatal(err) }

    str := string(b)

    for _, line := range strings.Split(str, "\n") {
        for i, char := range line {
            if unicode.IsNumber(char) {
                fmt.Printf("%d: %d\n", i, int(char - '0'))
            }
        }
    }
}
